From patchwork Thu Nov  8 10:00:58 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Janosch Frank <frankja@linux.ibm.com>
X-Patchwork-Id: 10673873
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 59B9C15E9
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  8 Nov 2018 10:01:18 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 427B22CAF7
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  8 Nov 2018 10:01:18 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 36A512CB2F; Thu,  8 Nov 2018 10:01:18 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C76BB2CAF7
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  8 Nov 2018 10:01:17 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726618AbeKHTf6 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 8 Nov 2018 14:35:58 -0500
Received: from mx0a-001b2d01.pphosted.com ([148.163.156.1]:57546 "EHLO
        mx0a-001b2d01.pphosted.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S1726421AbeKHTf6 (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 8 Nov 2018 14:35:58 -0500
Received: from pps.filterd (m0098396.ppops.net [127.0.0.1])
        by mx0a-001b2d01.pphosted.com (8.16.0.22/8.16.0.22) with SMTP id
 wA89sF8V079062
        for <kvm@vger.kernel.org>; Thu, 8 Nov 2018 05:01:15 -0500
Received: from e06smtp02.uk.ibm.com (e06smtp02.uk.ibm.com [195.75.94.98])
        by mx0a-001b2d01.pphosted.com with ESMTP id 2nmhf2c5vp-1
        (version=TLSv1.2 cipher=AES256-GCM-SHA384 bits=256 verify=NOT)
        for <kvm@vger.kernel.org>; Thu, 08 Nov 2018 05:01:15 -0500
Received: from localhost
        by e06smtp02.uk.ibm.com with IBM ESMTP SMTP Gateway: Authorized Use
 Only! Violators will be prosecuted
        for <kvm@vger.kernel.org> from <frankja@linux.ibm.com>;
        Thu, 8 Nov 2018 10:01:13 -0000
Received: from b06cxnps4074.portsmouth.uk.ibm.com (9.149.109.196)
        by e06smtp02.uk.ibm.com (192.168.101.132) with IBM ESMTP SMTP Gateway:
 Authorized Use Only! Violators will be prosecuted;
        (version=TLSv1/SSLv3 cipher=AES256-GCM-SHA384 bits=256/256)
        Thu, 8 Nov 2018 10:01:10 -0000
Received: from d06av23.portsmouth.uk.ibm.com (d06av23.portsmouth.uk.ibm.com
 [9.149.105.59])
        by b06cxnps4074.portsmouth.uk.ibm.com (8.14.9/8.14.9/NCO v10.0) with
 ESMTP id wA8A19rI7340358
        (version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-GCM-SHA384 bits=256
 verify=FAIL);
        Thu, 8 Nov 2018 10:01:09 GMT
Received: from d06av23.portsmouth.uk.ibm.com (unknown [127.0.0.1])
        by IMSVA (Postfix) with ESMTP id 19542A4051;
        Thu,  8 Nov 2018 10:01:09 +0000 (GMT)
Received: from d06av23.portsmouth.uk.ibm.com (unknown [127.0.0.1])
        by IMSVA (Postfix) with ESMTP id B9567A4053;
        Thu,  8 Nov 2018 10:01:08 +0000 (GMT)
Received: from frankja-ThinkPad-W541.fritz.box (unknown [9.145.28.214])
        by d06av23.portsmouth.uk.ibm.com (Postfix) with ESMTP;
        Thu,  8 Nov 2018 10:01:08 +0000 (GMT)
From: Janosch Frank <frankja@linux.ibm.com>
To: kvm@vger.kernel.org
Cc: linux-s390@vger.kernel.org, david@redhat.com, thuth@redhat.com,
        cohuck@redhat.com
Subject: [kvm-unit-tests PATCH] s390x: Fixup interrupt psws
Date: Thu,  8 Nov 2018 11:00:58 +0100
X-Mailer: git-send-email 2.7.4
In-Reply-To: <1541669224-29953-1-git-send-email-frankja@linux.ibm.com>
References: <1541669224-29953-1-git-send-email-frankja@linux.ibm.com>
X-TM-AS-GCONF: 00
x-cbid: 18110810-0008-0000-0000-0000028F08C4
X-IBM-AV-DETECTION: SAVI=unused REMOTE=unused XFE=unused
x-cbparentid: 18110810-0009-0000-0000-000021F94C10
Message-Id: <1541671258-4294-1-git-send-email-frankja@linux.ibm.com>
X-Proofpoint-Virus-Version: vendor=fsecure engine=2.50.10434:,,
 definitions=2018-11-08_06:,,
 signatures=0
X-Proofpoint-Spam-Details: rule=outbound_notspam policy=outbound score=0
 priorityscore=1501
 malwarescore=0 suspectscore=1 phishscore=0 bulkscore=0 spamscore=0
 clxscore=1015 lowpriorityscore=0 mlxscore=0 impostorscore=0
 mlxlogscore=449 adultscore=0 classifier=spam adjust=0 reason=mlx
 scancount=1 engine=8.0.1-1807170000 definitions=main-1811080087
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Let's use the svc psw instead of the mccheck psw for svc interrupts
and name it supervisor call. Also we exchange the io int psw for the
right one.

Signed-off-by: Janosch Frank <frankja@linux.ibm.com>
Reviewed-by: Cornelia Huck <cohuck@redhat.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
Reviewed-by: Thomas Huth <thuth@redhat.com>
---
 lib/s390x/interrupt.c | 2 +-
 s390x/cstart64.S      | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/s390x/interrupt.c b/lib/s390x/interrupt.c
index 6d0d4a3..4bcfdd6 100644
--- a/lib/s390x/interrupt.c
+++ b/lib/s390x/interrupt.c
@@ -137,6 +137,6 @@ void handle_io_int(void)
 
 void handle_svc_int(void)
 {
-	report_abort("Unexpected service call interrupt: at %#lx",
+	report_abort("Unexpected supervisor call interrupt: at %#lx",
 		     lc->svc_old_psw.addr);
 }
diff --git a/s390x/cstart64.S b/s390x/cstart64.S
index d6d40b7..dc7ddd6 100644
--- a/s390x/cstart64.S
+++ b/s390x/cstart64.S
@@ -33,10 +33,10 @@ init_psw_cont:
 	larl	%r1, mcck_int_psw
 	mvc	GEN_LC_MCCK_NEW_PSW(16), 0(%r1)
 	/* setup io interrupt handler */
-	larl	%r1, ext_int_psw
+	larl	%r1, io_int_psw
 	mvc	GEN_LC_IO_NEW_PSW(16), 0(%r1)
 	/* setup svc interrupt handler */
-	larl	%r1, mcck_int_psw
+	larl	%r1, svc_int_psw
 	mvc	GEN_LC_SVC_NEW_PSW(16), 0(%r1)
 	/* setup cr0, enabling e.g. AFP-register control */
 	larl	%r1, initial_cr0
